File: backend/app/routers/items.py
# app/routers/items.py
from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy.ext.asyncio import AsyncSession
from ..database import get_db
from .. import schemas, crud

router = APIRouter(prefix="/items", tags=["items"])

@router.get("", response_model=list[schemas.ItemRead])
async def list_items(db: AsyncSession = Depends(get_db)):
    return await crud.list_items(db)

@router.post("", response_model=schemas.ItemRead)
async def create_item(payload: schemas.ItemCreate, db: AsyncSession = Depends(get_db)):
    return await crud.create_item(db, payload)

# приклад збереження старих маршрутів, якщо були:
@router.get("/{item_id}", response_model=schemas.ItemRead)
async def get_item(item_id: int, db: AsyncSession = Depends(get_db)):
    obj = await crud.get_item(db, item_id)
    if not obj:
        raise HTTPException(404, "Item not found")
    return obj

@router.patch("/{item_id}", response_model=schemas.ItemRead)
async def update_item(item_id: int, payload: schemas.ItemUpdate, db: AsyncSession = Depends(get_db)):
    obj = await crud.update_item(db, item_id, payload)
    if not obj:
        raise HTTPException(404, "Item not found")
    return obj

File: frontend/src/api/items.ts
import { api } from "./client";

export type Item = { id: number; title: string; description: string };
export type ItemCreate = { title: string; description?: string };
export type ItemUpdate = { title?: string; description?: string };

export const listItems = async (): Promise<Item[]> => {
  const { data } = await api.get("/items");
  return data;
};

export const createItem = async (payload: ItemCreate): Promise<Item> => {
  const { data } = await api.post("/items", payload);
  return data;
};

export const updateItem = async (id: number, payload: ItemUpdate): Promise<Item> => {
  const { data } = await api.patch(`/items/${id}`, payload);
  return data;
};

export const deleteItem = async (id: number): Promise<void> => {
  await api.delete(`/items/${id}`);
};
